packages/static-hosting/lib/static-hosting.ts

```diff
@@ -1,144 +1,99 @@
   defaultRootObject?: string;
-  enforceSSL?: boolean;
-  /**
-   * Disable the use of the CSP header. Default value is false.
-   */
-  disableCSP?: boolean;
-  /**
-   * AWS limits the max header size to 1kb, this is too small for complex csp headers.
-   * The main purpose of this csp header is to provide a method of setting a report-uri.
-   */
+  certificateArn: string;
+  defaultBehaviorEdgeLambdas: EdgeLambda[];
+  additionalBehaviors?: Record<string, BehaviorOptions>;
+  enableErrorConfig?: boolean;
+  createDnsRecord?: boolean;
+  zoneName?: string;
   csp?: CSP;
-  /**
-   * This will generate a csp based *purely* on the provided csp object.
-   * Therefore disabling the automatic adding of common use-case properties.
-   */
   explicitCSP?: boolean;
+  responseHeadersPolicies?: ResponseHeaderMappings;
 
   /**
-   * Extend the default props for S3 bucket
-   */
-  s3ExtendedProps?: BucketProps;
-
-  /**
-   * Optional WAF ARN
-   */
-  webAclArn?: string;
-
-  /**
-   * Response Header policies
+   * After switching constructs, you need to maintain the same logical ID
+   * for the underlying CfnDistribution if you wish to avoid the deletion
+   * and recreation of your distribution.
+   *
+   * To do this, use escape hatches to override the logical ID created by
+   * the new Distribution construct with the logical ID created by the
+   * old construct
+   *
+   * https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cloudfront-readme.html#migrating-from-the-original-cloudfrontwebdistribution-to-the-newer-distribution-construct.
    */
-  responseHeadersPolicies?: ResponseHeaderMappings[];
-}
-
-interface remapPath {
-  from: string;
-  to: string;
+  overrideLogicalId?: string;
 }
 
 export interface ResponseHeaderMappings {
-  header: ResponseHeadersPolicy;
-  pathPatterns: string[];
-  attachToDefault?: boolean;
+  defaultBehaviorResponseHeaderPolicy?: ResponseHeadersPolicy;
+  additionalBehaviorResponsePolicy?: Record<string, ResponseHeadersPolicy>;
 }
 
-export class StaticHosting extends Construct {
-  private staticFiles = [
-    "js",
-    "css",
-    "json",
-    "svg",
-    "jpg",
-    "jpeg",
-    "png",
-    "gif",
-    "ico",
-    "woff",
-    "woff2",
-    "otf",
-  ];
+type Writeable<T> = { -readonly [P in keyof T]: T[P] };
 
+export class StaticHosting extends Construct {
   constructor(scope: Construct, id: string, props: StaticHostingProps) {
     super(scope, id);
 
-    // Should the stackExportPrefix is empty, 'StaticHosting' should be used as the prefix
     const exportPrefix = props.exportPrefix
       ? props.exportPrefix
       : "StaticHosting";
 
-    const siteName = `${props.subDomainName}.${props.domainName}`;
-    const siteNameArray: Array<string> = [siteName];
+    const siteName = props.domainNames[0];
     const enforceSSL = props.enforceSSL !== false;
-    const enableStaticFileRemap = props.enableStaticFileRemap !== false;
     const disableCSP = props.disableCSP === true;
```

Comment:
This was removed from props so it'll always be false right?

Reply:
The option still exists, see other comment. I'll mark this as resolved.